### PR TITLE
Clean up all TracerProviders registered using AddOpenTelemetryTracing

### LINKED ISSUE #1215: Extensions.Hosting can add multiple TracerProvider without disposing old
A code like below will add 2 instances of TracerProvider and new ones replace the old one. Old ones are not disposed when being overwritten.

```
var builder = new HostBuilder().ConfigureServices(services =>
            {
                services.AddOpenTelemetryTracing((builder) => builder.SetResource(Resources.Resources.CreateServiceResource("mys1")));
                services.AddOpenTelemetryTracing((builder) => builder.SetResource(Resources.Resources.CreateServiceResource("mys2")));
            });
```

Opening an issue to track as this was reported offline.

### PR DESCRIPTION
**Pull request by CodeBlanch**
Fixes #1215
Related to #2000

### CODE REVIEW

File: src/OpenTelemetry.Extensions.Hosting/Implementation/TelemetryHostedService.cs [hunk 1]
```diff
@@ -41,7 +42,7 @@ public Task StartAsync(CancellationToken cancellationToken)
                 // This method is invoked when host starts, and
                 // by requesting the TracerProvider from DI
                 // it ensures all instrumentations gets started.
-                this.serviceProvider.GetRequiredService<TracerProvider>();
+                this.serviceProvider.GetRequiredService<IEnumerable<TracerProvider>>();
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Extensions.Hosting/Implementation/TelemetryHostedService.cs:45]: @cijothomas This is what I was saying on the SIG meeting today. Really super easy to support multiple TracerProviders registered through DI. Just asking the container here for all of the registrations will cause them all to spin up, and then they will all dispose when the container goes down (test now passes for that reason). 

If we don't want to make it easy, we could check the IEnumerable here and throw if more than one is found?

**Review by reyang — APPROVED** [review]
LGTM.

File: src/OpenTelemetry.Extensions.Hosting/Implementation/TelemetryHostedService.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Extensions.Hosting/Implementation/TelemetryHostedService.cs:45]: While this solves the bug of not disposing a provider, I think this introduces another major issue:

```
var builder = new HostBuilder().ConfigureServices(services =>
            {
                services.AddOpenTelemetryTracing(builder =>
                {
                    builder.AddInstrumentation(() => testInstrumentation1);
                    builder.SetSampler<ParentBasedSampler>();
                });

                services.AddOpenTelemetryTracing(builder =>
                {
                    builder.AddInstrumentation(() => testInstrumentation2);
                    builder.SetSampler<ParentBasedSampler>();
                });

                // Which TracerProvider instance is being affected with the below line?
                services.AddSingleton<ParentBasedSampler>(new ParentBasedSampler(new AlwaysOffSampler()));
            });
```

**Review by cijothomas — CHANGES_REQUESTED** [review]
We want to understand the implications of having multiple providers added to DI. It could be a point of confusion for users.

File: src/OpenTelemetry.Extensions.Hosting/Implementation/TelemetryHostedService.cs (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Extensions.Hosting/Implementation/TelemetryHostedService.cs:45]: In that case, they both get the same one. The user has configured two TracerProviders. They differ by instrumentation but use the same sampler instance. I don't see a bug there, working as designed IMO,

**Comment by CodeBlanch** [comment]
I actually think with this PR, what we have works really well for the common case (one-provider), and also the case of users wanting to have more than one provider. The use case on #1998 is less clear to me. I'll add some comments on there with more thoughts.

**Pull request closed without merging**